R/kenwardroger.R

```diff
@@ -0,0 +1,160 @@
+#' obtain Kenward-Roger adjustment component
+#' @param tmb_data (`mmrm_tmb_data`)\cr produced by [h_mmrm_tmb_data()].
+#' @param theta (`numeric`)\cr numeric vector of the theta estimate.
+#' @details the function returns a named list, \eqn{P}, \eqn{Q} and \eqn{R}, which corresponds to the
+#' paper in 1997. The matrices are stacked in columns so that \eqn{P}, \eqn{Q} and \eqn{R} has the same
+#' column number(number of beta parameters). The number of rows, is dependent on
+#' the total number of theta and number of groups, if the fit is a grouped mmrm.
+#' For \eqn{P} matrix, it is stacked sequentially. For \eqn{Q} and \eqn{R} matrix, it is stacked so
+#' that the \eqn{Q_{ij}} and \eqn{R_{ij}} is stacked from \eqn{j} then to \eqn{i}, i.e. \eqn{R_{i1}}, \eqn{R_{i2}}, etc.
+#' \eqn{Q} and \eqn{R} only contains intra-group results and inter-group results should be all zero matrices
+#' so they are not stacked in the result.
+#' @return Named list with elements:
+#' - `P`: `matrix` of P component.
```

Comment:
```suggestion
#' - `P`: `matrix` of \eqn{P} component.
```
etc below to use same style as above